packages/api/src/tasks/checkAlerts.ts

```diff
@@ -160,27 +160,144 @@ export const notifyChannel = async ({
             }),
       });
 
-      // ONLY SUPPORTS SLACK WEBHOOKS FOR NOW
       if (webhook?.service === 'slack') {
-        await slack.postMessageToWebhook(webhook.url, {
-          text: message.title,
-          blocks: [
-            {
-              type: 'section',
-              text: {
-                type: 'mrkdwn',
-                text: `*<${message.hdxLink} | ${message.title}>*\n${message.body}`,
-              },
-            },
-          ],
-        });
+        await handleSendSlackWebhook(webhook, message);
+      } else if (webhook?.service === 'generic') {
+        await handleSendGenericWebhook(webhook, message);
       }
       break;
     }
     default:
       throw new Error(`Unsupported channel type: ${channel}`);
   }
 };
+
+const handleSendSlackWebhook = async (
+  webhook: IWebhook,
+  message: {
+    hdxLink: string;
+    title: string;
+    body: string;
+  },
+) => {
+  if (!webhook.url) {
+    throw new Error('Webhook URL is not set');
+  }
+
+  await slack.postMessageToWebhook(webhook.url, {
+    text: message.title,
+    blocks: [
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: `*<${message.hdxLink} | ${message.title}>*\n${message.body}`,
+        },
+      },
+    ],
+  });
+};
+
+export const handleSendGenericWebhook = async (
+  webhook: IWebhook,
+  message: {
+    hdxLink: string;
+    title: string;
+    body: string;
+  },
+) => {
+  // QUERY PARAMS
+
+  if (!webhook.url) {
+    throw new Error('Webhook URL is not set');
+  }
+
+  let url: string;
+  // user input of queryParams is disabled on the frontend for now
+  if (webhook.queryParams) {
+    const queryParamsString = new URLSearchParams(
+      webhook.queryParams,
+    ).toString();
+
+    // user may have included params in both the url and the query params
+    // so they should be merged
+    const tmpURL = new URL(webhook.url);
+    if (queryParamsString) {
+      tmpURL.search = tmpURL.search
+        ? `${tmpURL.search}&${queryParamsString}`
+        : queryParamsString;
```

Comment:
nit: I wonder if we can use `tmpURL.searchParams.append(xxx, yyy)` (since queryParams is an object)

Reply:
oh for sure, that was back when it was a string, then we pulled QPs out for now and I didn't refactor - bunch of extra steps there 👍 ill adjust